src/pane.coffee

```diff
@@ -550,14 +550,17 @@ class Pane extends Model
   destroyItem: (item) ->
     index = @items.indexOf(item)
     if index isnt -1
+      @setActiveItem(item)
       @emitter.emit 'will-destroy-item', {item, index}
       @container?.willDestroyPaneItem({item, index, pane: this})
-      if @promptToSaveItem(item)
-        @removeItem(item, false)
-        item.destroy?()
-        true
-      else
-        false
+      promptToSaveItemAsync = ()=>
+        if @promptToSaveItem(item)
+          @removeItem(item, false)
+          item.destroy?()
+          true
+        else
+          false
+      setTimeout promptToSaveItemAsync, 100
```

Comment:
`@applicationDelegate.confirm` is sync call and it is freeze execution. But setActiveItem uses emitter (`@emitter.emit 'did-change-active-item', @activeItem`), so it is called asynchronously. As a result prompt get showed before switch to active tab happens and then execution freeze. To resolve this need to make `@applicationDelegate.confirm` async. But `promptToSaveItem` is treated as sync in `confirmClose` and `destroyItem`. So to make `promptToSaveItem` we also will need to make async dependent functions. This would be easy with Promises and `await`. But `await` is not available in coffeescript, see https://github.com/jashkenas/coffeescript/pull/1942 and https://github.com/jashkenas/coffeescript/pull/3757